queue/bytes_queue.go

```diff
@@ -199,32 +204,45 @@ func (q *BytesQueue) peekCheckErr(index int) error {
 		return errInvalidIndex
 	}
 
-	if index+headerEntrySize >= len(q.array) {
+	if index >= len(q.array) {
 		return errIndexOutOfBounds
 	}
 	return nil
 }
 
-func (q *BytesQueue) peek(index int) ([]byte, int, error) {
+func (q *BytesQueue) peek(index int) ([]byte, int, int, error) {
 	err := q.peekCheckErr(index)
 	if err != nil {
-		return nil, 0, err
+		return nil, 0, 0, err
 	}
 
-	blockSize := int(binary.LittleEndian.Uint32(q.array[index : index+headerEntrySize]))
-	return q.array[index+headerEntrySize : index+headerEntrySize+blockSize], blockSize, nil
+	blockSize, n := binary.Uvarint(q.array[index:])
+	if n <= 0 {
+		panic("wrong encoding")
```

Comment:
Hm, should we panic here? simple `return nil, 0, 0, err` should be enough, isn't it?

Reply:
Yes, do you have any suggestions on which error should it be?